rslib/src/import_export/package/apkg/import.rs

```diff
@@ -248,12 +255,23 @@ impl<'a> Context<'a> {
         let notetype = self.get_expected_notetype(note.notetype_id)?;
         note.prepare_for_update(&notetype, self.normalize_notes)?;
         note.usn = self.usn;
-        let old_id = std::mem::take(&mut note.id);
-        self.target_col.add_note_only_undoable(note)?;
-        self.remapped_notes.insert(old_id, note.id);
+        self.uniquify_note_id(note);
+
+        self.target_col.add_note_only_with_id_undoable(note)?;
+        self.existing_notes.insert(note.id);
         Ok(())
     }
 
+    fn uniquify_note_id(&mut self, note: &mut Note) {
+        let original = note.id;
+        while self.existing_notes.contains(&note.id) {
+            note.id.0 += 999;
```

Comment:
At a later stage when this code is functional, it might be interesting to benchmark this approach compared to a simple call to storage.add_note(), which assigns the next-greatest available id in the case of a conflict. The latter is a bit easier to reason about, but if it introduces a significant performance regression, then we're better off sticking with what we've got.

Reply:
Isn't the problem with assigning the next available id that it messes up note creation time?

Reply:
Yes, you're quite right, scratch that idea.